src/extractor/guidance/intersection_generator.cpp

```diff
@@ -158,7 +162,134 @@ Intersection IntersectionGenerator::getConnectedRoads(const NodeID from_node,
         intersection[self_u_turn].entry_allowed = true;
     }
 
-    return mergeSegregatedRoads(std::move(intersection));
+    return intersection;
+}
+
+bool IntersectionGenerator::CanMerge(const NodeID node_at_intersection,
```

Comment:
See https://github.com/Project-OSRM/osrm-backend/blob/90cc782b4581f25f3350f5c0df91c3de45e59625/include/extractor/guidance/intersection_generator.hpp#L64-L81. Can move the explanation here, if you feel this is the better location.